django/thunderstore/frontend/api/experimental/tests/test_community_package_list.py

```diff
@@ -0,0 +1,418 @@
+from typing import Dict, List, Union
+
+import pytest
+from django.urls import reverse
+from rest_framework.test import APIClient
+
+from thunderstore.cache.cache import CacheBustCondition, bust_cache_by_condition
+from thunderstore.community.factories import CommunitySiteFactory, PackageListingFactory
+from thunderstore.community.models import (
+    PackageCategory,
+    PackageListingReviewStatus,
+    PackageListingSection,
+)
+from thunderstore.community.models.package_listing import PackageListing
+from thunderstore.frontend.api.experimental.views import CommunityPackageListApiView
+from thunderstore.repository.factories import (
+    PackageRatingFactory,
+    PackageVersionFactory,
+)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def clear_pagination_cache():
+    bust_cache_by_condition(CacheBustCondition.any_package_updated)
+
+
+@pytest.mark.django_db
+def test_only_packages_listed_in_community_are_returned(api_client: APIClient) -> None:
+    listing1 = PackageListingFactory()
+    PackageListingFactory()
+
+    data = __query_api(api_client, listing1.community.identifier)
+
+    __assert_packages_by_listings(data, listing1)
+
+
+@pytest.mark.django_db
+def test_only_active_packages_are_returned(api_client: APIClient) -> None:
+    listing1 = PackageListingFactory(package_kwargs={"is_active": False})
+    PackageListingFactory()
+
+    data = __query_api(api_client, listing1.community.identifier)
+
+    assert len(data["packages"]) == 0
+
+
+@pytest.mark.django_db
+def test_only_community_listings_for_correct_community_are_prefetched(
```

Comment:
How does this actually test it's a prefetch that happens, or does it not do that and the name is misleading?

Query count assertions can be done if found useful, there are some example tests with those.

Reply:
Renamed the test from `...are_prefetched` to `...are_included_in_queryset` to better communicate that this test case only tests the view's `get_queryset` method rather than HTTP response of the view.